src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -1877,10 +1875,7 @@ WorkerAggregateExpressionList(Aggref *originalAggregate,
 		CountDistinctErrorRate == DISABLE_DISTINCT_APPROXIMATION &&
 		walkerContext->repartitionSubquery)
 	{
-		Aggref *aggregate = (Aggref *) copyObject(originalAggregate);
-		List *aggTargetEntryList = aggregate->args;
-		TargetEntry *distinctTargetEntry = (TargetEntry *) linitial(aggTargetEntryList);
-		List *columnList = pull_var_clause_default((Node *) distinctTargetEntry);
+		List *columnList = pull_var_clause_default((Node *) originalAggregate);
```

Comment:
Just a tiny difference here is that the code used to `copy` the original aggregate, so using copied `Vars`. Now, it uses the original `Vars`. But, below we already copy the columns when necessary. If you also think that it's OK, this is a :shipit: